### PR TITLE
[spec] Add specification for May 2020 Upgrade

### PR DESCRIPTION
**Pull request by Mengerian**

### CODE REVIEW

File: spec/2020-05-15-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+The following are not consensus changes, but are recommended policy changes for Bitcoin Cash implementations:
+
+* The default for max number of in-mempool ancestors is changed from 25 to 50.
+* The default for max number of in-mempool descendants is changed from 25 to 50.
+* Automatic replay protection for future upgrade.
+
+## SigChecks
+
+Enforcement sigops limits is removed, and replaced with new limits based on the number of signature checks that are actually executed when running a script. This new system is called SigChecks.
```
Comment (jasonbcox) [inline spec/2020-05-15-upgrade.md]: Enforcement `of`? sigops limits is removed

File: spec/2020-05-15-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+Details can be found in the [full specification: SigChecks](https://github.com/bitcoincashorg/bitcoincash.org/pull/443).
```
Comment (jasonbcox) [inline spec/2020-05-15-upgrade.md]: Update the link: https://github.com/bitcoincashorg/bitcoincash.org/blob/master/spec/2020-05-15-sigchecks.md

File: spec/2020-05-15-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+## OP_REVERSEBYTES
+
+This new opcode reverses the order of bytes in a string. It can be used to change endianness.
+
+Details can be found in the [full specification: OP_REVERSEBYTES](https://github.com/bitcoincashorg/bitcoincash.org/pull/358).
```
Comment (jasonbcox) [inline spec/2020-05-15-upgrade.md]: Update the link: https://github.com/bitcoincashorg/bitcoincash.org/blob/master/spec/2020-05-15-op_reversebytes.md

File: spec/2020-05-15-upgrade.md (hunk 1 above)
Comment (Mengerian) [reply spec/2020-05-15-upgrade.md]: OK, fixed

File: spec/2020-05-15-upgrade.md (hunk 2 above)
Comment (Mengerian) [reply spec/2020-05-15-upgrade.md]: done

File: spec/2020-05-15-upgrade.md (hunk 3 above)
Comment (Mengerian) [reply spec/2020-05-15-upgrade.md]: Done

**Comment by ftrader** [comment]
I consider the merge of this PR , sneaking in the addition of IP enforcement `cc21f831bd5f6be323723d82fd5d0998763a2703` which is not even well specified, and the merging of the PR by the same person who originated it, as something that should have never happened.

I will raise an issue and a pull request to fully revert the inclusion of the IFP component for May's upgrade - it does NOT have community consensus in any way.

As a personal note, I think this is disgraceful behavior by the stewards of this repository.

File: spec/2020-05-15-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## OP_REVERSEBYTES
+
+This new opcode reverses the order of bytes in a string. It can be used to change endianness.
+
+Details can be found in the [full specification: OP_REVERSEBYTES](https://github.com/bitcoincashorg/bitcoincash.org/blob/master/spec/2020-05-15-op_reversebytes.md).
+
+## Infrastructure Funding Plan
+
+The purpose of the Infrastructure Funding Plan (IFP) is to provide funding to development projects working on common Bitcoin Cash infrastructure. If activated, it enforces that 5% of the block reward is spent to one of a set of specified addresses. The IFP has the following features:
```
Comment (BigBlockIfTrue) [inline spec/2020-05-15-upgrade.md]: There is no agreement to include the IFP in the official Bitcoin Cash specification and we are already way past the feature-freeze deadline. NACK

**Comment by ShadowOfHarbringer** [comment]
Two issues:

1. It way past code freeze deadline

2. It is unacceptable for **any** single developer to re-define what "Bitcoin Cash" is without any kind of agreement among neither miners, community, developer or popular service providers.

This behavior is contemptible.

NACK.

**Comment by molecular** [comment]
NACK

**Pull request merged**